bbot/modules/lightfuzz/submodules/serial.py

```diff
@@ -0,0 +1,168 @@
+
+        2) If the first case doesn't match, we check for a telltale error string like "java.io.optionaldataexception" in the response.
+    """
+
+    friendly_name = "Unsafe Deserialization"
+
+    def is_possibly_serialized(self, value):
+        # Use the is_base64 method from BaseLightfuzz via self
+        if self.is_base64(value):
+            return True
+
+        # Use the is_hex method from BaseLightfuzz via self
+        if self.is_hex(value):
+            return True
+
+        # List of common PHP serialized data prefixes
+        php_serialized_prefixes = [
+            "a:",  # Array
+            "O:",  # Object
+            "s:",  # String
+            "i:",  # Integer
+            "d:",  # Double
+            "b:",  # Boolean
+            "N;",  # Null
+        ]
+
+        # Check if the value starts with any of the PHP serialized prefixes
+        if any(value.startswith(prefix) for prefix in php_serialized_prefixes):
+            return True
+        return False
+
+    async def fuzz(self):
+        cookies = self.event.data.get("assigned_cookies", {})
+        control_payload_hex = "f56124208220432ec767646acd2e6c6bc9622a62c5656f2eeb616e2f"
+        control_payload_base64 = "4Wt5fYx5Y3rELn5myS5oa996Ji7IZ28uwGdha4x6YmuMfG992CA="
+        control_payload_php_raw = "z:0:{}"
+        # These payloads are benign, no-op, or otherwise harmless
+        #  minimally sized valid serialized objects for their given language/platform
+        base64_serialization_payloads = {
+            "php_base64": "YTowOnt9",
+            "java_base64": "rO0ABXNyABFqYXZhLmxhbmcuQm9vbGVhbs0gcoDVnPruAgABWgAFdmFsdWV4cAA=",
+            "java_base64_string_error": "rO0ABXQABHRlc3Q=",
+            "java_base64_OptionalDataException": "rO0ABXcEAAAAAAEAAAABc3IAEGphdmEudXRpbC5IYXNoTWFwAAAAAAAAAAECAAJMAARrZXkxYgABAAAAAAAAAAJ4cHcBAAAAB3QABHRlc3Q=",
+            "dotnet_base64": "AAEAAAD/////AQAAAAAAAAAGAQAAAAdndXN0YXZvCw==",
+            "ruby_base64": "BAh7BjoKbE1FAAVJsg==",
+        }
+
+        hex_serialization_payloads = {
+            "java_hex": "ACED00057372000E6A6176612E6C616E672E426F6F6C65616ECD207EC0D59CF6EE02000157000576616C7565787000",
+            "java_hex_OptionalDataException": "ACED0005737200106A6176612E7574696C2E486173684D617000000000000000012000014C00046B6579317A00010000000000000278707000000774000474657374",
+            "dotnet_hex": "0001000000ffffffff01000000000000000601000000076775737461766f0b",
+        }
+
+        php_raw_serialization_payloads = {
+            "php_raw": "a:0:{}",
+        }
+
+        serialization_errors = [
+            "invalid user",
+            "cannot cast java.lang.string",
+            "dump format error",
+            "java.io.optionaldataexception",
+        ]
+
+        general_errors = [
+            "Internal Error",
+            "Internal Server Error",
+            "The requested URL was rejected",
+        ]
```

Comment:
As string constants these should be moved to the class level for performance, so they're not being constantly redefined

Reply:
fixed in 92d385130ca17e13b55972bc8b9c28d1e06b06c3